passthrough/usecase/passthrough_usecase.go

```diff
@@ -0,0 +1,140 @@
+package usecase
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	staking "github.com/cosmos/cosmos-sdk/x/staking/types"
+	concentratedLiquidity "github.com/osmosis-labs/osmosis/v25/x/concentrated-liquidity/client/queryproto"
+	lockup "github.com/osmosis-labs/osmosis/v25/x/lockup/types"
+
+	"github.com/osmosis-labs/sqs/domain/mvc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type passthroughUseCase struct {
+	poolsUseCase mvc.PoolsUsecase
+
+	bankQueryClient banktypes.QueryClient
+	stakingQueryClient staking.QueryClient
+	lockupQueryClient lockup.QueryClient
+	concentratedLiquidityQueryClient concentratedLiquidity.QueryClient
+}
+
+var _ mvc.PassthroughUsecase = &passthroughUseCase{}
+
+// NewPassThroughUsecase Creates a passthrough use case
+func NewPassThroughUsecase(grpcURI string, puc mvc.PoolsUsecase) (mvc.PassthroughUsecase, error){
+	grpcClient, err := grpc.Dial(grpcURI, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return nil, err
+	}
+	
+	return &passthroughUseCase{
+		poolsUseCase: puc,
+
+		bankQueryClient: banktypes.NewQueryClient(grpcClient),
+		stakingQueryClient: staking.NewQueryClient(grpcClient),
+		lockupQueryClient: lockup.NewQueryClient(grpcClient),
+		concentratedLiquidityQueryClient: concentratedLiquidity.NewQueryClient(grpcClient),
+	}, nil
+}
+
+func (p *passthroughUseCase) GetAccountCoinsTotal(ctx context.Context, address string) (sdk.Coins, error) {
```

Comment:
This should be split up into multiple different functions, one for each diff functionality, and get unit tested separately